### ISSUE TITLE
Replace usages of BlocklyMethods for C robots

### ISSUE THREAD

**Issue by boonto** [OpenRoberta/openroberta-lab#278]
Continuation issue to partially solved issues #95 and #230:
`BlocklyMethods` and `NEPODefs` still have to be removed from Java and C based robots.

- [x] Find all "BlocklyMethods.py" / "NEPOdefs.h" etc.
- [x] Find common C methods
- [x] Extract C one liners (constants?) directly into code generation
  - [x] Arduino based
  - [x] c4ev3
  - [x] Calliope
  - [x] NXT
- [x] Combine C math multi liners into common helper methods
  - [x] Arduino based
  - [x] c4ev3
  - [x] Calliope
  - [x] NXT
- [x] Implement class in all C robots
  - [x] Arduino based
  - [x] c4ev3
  - [x] Calliope
  - [x] NXT
- [x] Default implementations for C
- [x] Special implementations for C robots that don't support default
- [x] Unit tests
  - [x] Arduino based
  - [x] c4ev3
  - [x] Calliope
  - [x] NXT
- [x] Test some methods on robots
  - [x] Arduino based
  - [x] c4ev3
  - [x] Calliope
  - [x] NXT

**Comment by boonto** [comment]
We discussed some changes today.
The original goal was to simplify and show the code for everything from math to lists.
Now, for C specifically, we decided to create helper methods for math operations but omit the list operations (together with HAL-like abstractions, which should be moved into its own file at some point) in the `NEPODefs.h`. As these methods are mostly iterator based there is not a lot of didactic value to be gained.
